classes/submission/maps/Schema.php

```diff
@@ -362,10 +362,11 @@ public function getPropertyStages(Submission $submission): array
         $stageIds = Application::get()->getApplicationStages();
         $request = Application::get()->getRequest();
         $currentUser = $request->getUser();
-        $context = $request->getContext();
 
-        $stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /** @var StageAssignmentDAO $stageAssignmentDao */
-        $stageAssignments = $stageAssignmentDao->getBySubmissionAndUserIdAndStageId($submission->getId(), $currentUser->getId() ?? 0)->toArray();
+        // Replaces StageAssignmentDAO::getBySubmissionAndUserIdAndStageId
+        $stageAssignments = StageAssignment::withSubmissionId($submission->getId())
+            ->withUserId($currentUser->getId() ?? 0)
```

Comment:
Here I kept the same conditions as the [original](https://github.com/pkp/pkp-lib/blob/main/classes/submission/maps/Schema.php#L368) one. 

I think that this is done in order to return no `$stageAssignments` in the case that the `$currentUser` is not set. 

Because `$stageAssignments` is used below, after an if statement regarding the `$currentUser`, I transfered this retrieve call there.